### ISSUE TITLE
AOV plugin returns an exception

### ISSUE THREAD

**Issue by joeylitalien** [mitsuba-renderer/mitsuba3#900]
## Summary

AOV plugin is broken.

## System configuration

System information:

  OS: Ubuntu 20.04.6 LTS
  CPU: Intel(R) Core(TM) i7-9800X CPU @ 3.80GHz
  GPU: NVIDIA TITAN V
  Python: 3.10.13 (main, Aug 25 2023, 13:20:03) [GCC 9.4.0]
  NVidia driver: 510.39.01
  CUDA: 11.6.55
  LLVM: 12.0.0

  Dr.Jit: 0.4.3
  Mitsuba: 3.4.0
     Is custom build? False
     Compiled with: GNU 10.2.1
     Variants:
        scalar_rgb
        scalar_spectral
        cuda_ad_rgb
        llvm_ad_rgb

## Description

The AOV plugin returns a critical exception in all RGB variants.

## Steps to reproduce

1. Render a simple scene with `aov` integrator as stated in the docs:
```
<scene version="2.1.0">
    <default name="spp" value="16"/>
    <default name="resx" value="512"/>
    <default name="resy" value="512"/>

    <integrator type="aov">
        <string name="aovs" value="d:depth,nn:sh_normal"/>
        <integrator type="path" name="my_image"/>
    </integrator>

    <sensor type="perspective">
        <float name="fov" value="30"/>
        <transform name="to_world">
            <lookat origin="5, 5, 0" target="0, 0, 0" up="0, 1, 0"/>
        </transform>
        <sampler type="independent">
            <integer name="sample_count" value="$spp"/>
        </sampler>
        <film type="hdrfilm">
            <integer name="width" value="$resx"/>
            <integer name="height" value="$resy"/>
        </film>
    </sensor>

    <emitter type="constant">
        <rgb name="radiance" value="1.0"/>
    </emitter>

    <shape type="sphere">
        <bsdf type="diffuse">
            <rgb name="reflectance" value="0.5"/>
        </bsdf>
    </shape>
</scene>
```

Somehow this throws the error: `Caught a critical exception: Tensor::shape(2): out of bounds!`.

**Comment by rtabbara** [comment]
Hi @joeylitalien ,

Thanks for reporting this. We've tracked down the issue and a PR should hopefully be in soon

**Comment by rtabbara** [comment]
Fixed in #901 and part of 3.4.1 release
